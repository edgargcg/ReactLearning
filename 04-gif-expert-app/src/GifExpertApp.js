import { Fragment, useState } from "react";
import { CategoryAdd } from "./Components/Categories/CategoryAdd";
import { GifGrid } from "./Components/Gifs/GifGrid";

const GifExpertApp = ( { defaultCategories = [] } ) => {
    const [categories, setCategories] = useState(defaultCategories)

    const handleAddCategory = (value) => {
        if (!categories.find( c => c === value))
            setCategories([value, ...categories]);
    }

    return (
        <Fragment>
            <div className="container">
                <h2 className="title">
                    GiftExpertApp
                </h2>
                <CategoryAdd setCategory={ handleAddCategory } />
                <div className="categories">    
                    {
                        categories.map(
                            category => 
                                (
                                    <GifGrid 
                                        key={category}
                                        category={category} 
                                    />
                                )
                        )
                    }
                </div>
            </div>
        </Fragment>
    );
}

export default GifExpertApp;