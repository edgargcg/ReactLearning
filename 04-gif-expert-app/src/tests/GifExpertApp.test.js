import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Test on <GifExpertApp />', () => {
    const wrapper = shallow(<GifExpertApp />)

    test('should render <GifExpertApp />', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should render a list of categories', () => {
        const categories = ['One punch man', 'Dragon Ball'];
        const wrapper = shallow( <GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
})
