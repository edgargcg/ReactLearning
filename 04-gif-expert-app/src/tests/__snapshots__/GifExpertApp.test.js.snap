// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifExpertApp /> should render <GifExpertApp /> 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h2
      className="title"
    >
      GiftExpertApp
    </h2>
    <CategoryAdd
      setCategory={[Function]}
    />
    <div
      className="categories"
    />
  </div>
</Fragment>
`;

exports[`Test on <GifExpertApp /> should render a list of categories 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h2
      className="title"
    >
      GiftExpertApp
    </h2>
    <CategoryAdd
      setCategory={[Function]}
    />
    <div
      className="categories"
    >
      <GifGrid
        category="One punch man"
        key="One punch man"
      />
      <GifGrid
        category="Dragon Ball"
        key="Dragon Ball"
      />
    </div>
  </div>
</Fragment>
`;
